iib/web/models.py

```diff
@@ -0,0 +1,219 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+from enum import Enum
+
+import sqlalchemy
+
+from iib.exceptions import ValidationError
+from iib.web import db
+
+
+class BaseEnum(Enum):
+    """A base class for IIB enums."""
+
+    @classmethod
+    def get_names(cls):
+        """
+        Get a sorted list of enum names.
+
+        :return: a sorted list of valid enum names
+        :rtype: list
+        """
+        return sorted([e.name for e in cls])
+
+
+class RequestStateMapping(BaseEnum):
+    """An Enum that represents the request states."""
+
+    in_progress = 1
+    complete = 2
+    failed = 3
+
+
+class RequestTypeMapping(BaseEnum):
+    """An Enum that represents the request types."""
+
+    add = 1
+    rm = 2
+
+
+class Architecture(db.Model):
+    """
+    An architecture associated with an image.
+    """
+
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String, nullable=False, unique=True)
+
+    def __repr__(self):
+        return '<Architecture name={0!r}>'.format(self.name)
+
+
+class ImageArchitecture(db.Model):
+    """
+    An association table between images and the architectures they were built for.
+
+    This will only be used for images built by IIB.
+    """
+
+    # A primary key is required by SQLAlchemy when using declaritive style tables, so a composite
+    # primary key is used on the two required columns
+    image_id = db.Column(
+        db.Integer, db.ForeignKey('image.id'), autoincrement=False, index=True, primary_key=True
```

Comment:
If both `image_id` and `architecture_id` are primary keys, then won't this enforce a one-to-one relationship between Image and Architecture? 

IIUC, this should be a many-to-many relationship.

Reply:
A unit test to validate this behavior would be nice.

Reply:
The primary key is used to uniquely identify a row. Since this is an association table that maps images to architectures, there would not be more than one row where `image_id` and `architecture_id` are the same.

As per your unit test comment. I think you were asking me to modify the existing unit test so that more than one architecture was added to the image. I updated the test to do this in the latest push.

Reply:
Thank you for the explanation. The test does help clear things up.